server/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/daco/DaCoMaidProcessor.java

```diff
@@ -0,0 +1,215 @@
+        if (dataEntry.find()) {
+          String name = dataEntry.group("entryName");
+          String lvl = dataEntry.group("lvl");
+          if (name != null
+                  && !VARIABLE_LEVEL_66.equals(lvl)
+                  && !VARIABLE_LEVEL_77.equals(lvl)
+                  && !VARIABLE_LEVEL_88.equals(lvl)) {
+            lastSuffix = DaCoHelper.extractSuffix(name);
+          }
+        }
+      } else if (dataDivisionPattern.matcher(line).find()) {
+        state = DaCoMaidProcessingState.DATA_DIVISION;
+      }
+      lines[i] = collectCopyMaid(line, i, copyMaidNodes, lastSuffix, context, errors);
+    }
+    input = String.join("\n", lines);
+
+
+    return new DialectOutcome(input, copyMaidNodes);
+  }
+
+  private String collectCopyMaid(String input, int lineNumber, List<Node> copyMaidNodes, String lastSuffix, DialectProcessingContext context, List<SyntaxError> errors) {
+    Matcher matcher = copyMaidPattern.matcher(input);
+
+    if (matcher.find()) {
+      StringBuffer sb = new StringBuffer();
+      String indent = matcher.group("indent");
+      int startChar = indent == null ? 0 : matcher.end("indent");
+      int endChar = matcher.end(matcher.groupCount() - 1);
+      int len = endChar - startChar;
+      matcher.appendReplacement(sb, (indent == null ? "" : indent)
+              + String.join("", Collections.nCopies(len, CobolDialect.FILLER)));
+      String level = matcher.group("level");
+      String layoutId = matcher.group("layoutId");
+      String layoutUsage = matcher.group("layoutUsage");
+      if (level != null) {
+        Range range = new Range(new Position(lineNumber, matcher.start("layoutId")), new Position(lineNumber, matcher.end("layoutId")));
+        copyMaidNodes.add(
+                createMaidCopybookNode(context, Integer.parseInt(level), layoutId, layoutUsage, lastSuffix, range, errors)
+        );
+      }
+      matcher.appendTail(sb);
+      return sb.toString();
+    }
+    return input;
+  }
+
+
+  private CopyNode createMaidCopybookNode(DialectProcessingContext context, int startingLevel, String layoutId, String layoutUsage, String lastSuffix, Range range, List<SyntaxError> errors) {
+    Locality locality = Locality.builder()
+            .uri(context.getProgramDocumentUri())
+            .range(range)
+            .build();
+    DaCoCopyNode cbNode = new DaCoCopyNode(locality, makeCopybookFileName(layoutId, layoutUsage), layoutUsage, lastSuffix);
+
+    CopybookName copybookName = new CopybookName(makeCopybookFileName(layoutId, layoutUsage), DaCoDialect.NAME);
+    CopybookModel copybookModel = copybookService.resolve(copybookName,
+            context.getProgramDocumentUri(),
+            context.getProgramDocumentUri(),
+            context.getCopybookConfig(),
+            true);
+    if (copybookModel.getContent() != null) {
+      Location location = new Location(copybookModel.getUri(),
+              new Range(new Position(0, 0), new Position(0, 0)));
```

Comment:
Am not sure. But should the definition range be of the entire copybook content? or is this ok, to locate it at the beginning of the file.